mmv1/templates/terraform/pre_update/alloydb_instance.go.erb

```diff
@@ -0,0 +1,7 @@
+// Avoid modifying instance type. No use case for this
+
+// Secondary instances can not be created in primary cluster and vice versa.
+// Default change is destroy before create, so modifcation from primary <-> secondary will only delete the instance with failure to create the instance.
+if d.HasChange("instance_type") {
+    return fmt.Errorf("Can not change instance type after creation.")
+}
```

Comment:
Why not simply make `instance_type` immutable ?

Reply:
instance_type is already immutable.
And since it is immutable, if a user changes the type of instance from PRIMARY to SECONDARY or vice versa then terraform tries to delete the resource first and then tries to create it.
And since a primary(secondary) instance can not be created in a secondary(primary) cluster, it would delete the instance successfully and fail to create the instance.

Moreover, there is no use case where a user would need to change the instance_type.

Reply:
but update won't be called if instance type is changed. Only create and delete are called for immutable fields. So we should be able to omit this pre_update step.

Reply:
From the terraform behavior that I am seeing with `instance_type: immutable`
If a user updates the instance_type from PRIMARY to SECONDARY, terraform plans to destroy the PRIMARY instance and create a SECONDARY instance. Since this is not an expected behavior or serves any use case, a user can explicitly add   
`lifecycle {
    ignore_changes = [instance_type]
}`
to stop this from happening.
Or we can give a user friendly error, so the system is robust even if the user forgets to specify the lifecyle.ignore_change field in their config.